src/panels/lovelace/cards/hui-shopping-list-card.ts

```diff
@@ -250,6 +301,49 @@ class HuiShoppingListCard extends SubscribeMixin(LitElement)
     }
   }
 
+  private async _toggleReorder() {
+    if (!Sortable) {
+      const sortableImport = await import(
+        "sortablejs/modular/sortable.core.esm"
+      );
+      Sortable = sortableImport.Sortable;
+    }
+    this._reordering = !this._reordering;
+    await this.updateComplete;
+    if (this._reordering) {
+      this._createSortable();
+    } else {
+      this._sortable?.destroy();
+      this._sortable = undefined;
+    }
+  }
+
+  private _createSortable() {
+    this._sortable = new Sortable(this.shadowRoot!.getElementById("sortable"), {
+      animation: 150,
+      fallbackClass: "sortable-fallback",
+      dataIdAttr: "item-id",
+      handle: "ha-svg-icon",
+      onSort: async (evt) => {
+        reorderItems(this.hass!, this._sortable.toArray()).catch(() =>
+          this._fetchData()
```

Comment:
We should let the user know it failed and why we undo all the sorting work.

Reply:
I will add a toast for the error message tonight.